src/lib/components/HarnessSelector/HarnessSelector.svelte

```diff
@@ -57,13 +65,26 @@
 
   const handleClear = () => {
     // clear search input
-    inputValue = "";
-    handleInput();
-    inputRef?.focus();
-
-     // clear harness selection
-     selection = null;
-     onChange(null);
+    let clearedInput = false;
+    if (inputValue) {
+      inputValue = "";
+      clearedInput = true;
+      handleInput();
+      inputRef?.focus();
+    }
+    // clear harness selection and close if we weren't clearing the search input
+    if (!clearedInput) {
+      // clear harness selection
+      if (selection) {
+        selection = null;
+        onChange(null);
+        updateQueryParams(null); // NOTE: Doing this causes a soft reload which removes the focus from the input
+      }
+      // close the dropdown if it's open
+      if (menuOpen) {
+        menuOpen = false;
+      }
+    }
```

Comment:
this might be a useful watch: https://www.youtube.com/watch?v=CFRhGnuXG-4

Reply:
I agree it's ugly, but my main concern was getting it to work properly and for whatever reason this is how it ended up. Upon closer look, I could have done an early return at least 🤔

But I'm glad you found a better way to do it. 😃